src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/GoalService.kt

```diff
@@ -66,9 +68,24 @@ class GoalService(
     val goalEntity = goalRepository.findByUuid(goalUuid)
       ?: throw Exception("This Goal was not found: $goalUuid")
 
-    goalEntity.title = goal.title
-    goalEntity.targetDate = goal.targetDate
-    goalEntity.status = if (goal.targetDate != null) GoalStatus.ACTIVE else GoalStatus.FUTURE
+    if (goal.title != null) {
+      goalEntity.title = goal.title
+    }
+
+    if (goal.targetDate != null) {
+      goalEntity.targetDate = goal.targetDate
+      if (goalEntity.status == GoalStatus.FUTURE) {
+        goalEntity.status = GoalStatus.ACTIVE
+      }
+    }
+
+    if (goal.targetDate == null && goal.status == GoalStatus.FUTURE) {
+      goalEntity.targetDate = null
+    }
+
+    if (goal.status != null) {
+      goalEntity.status = goal.status
+    }
 
     var relatedAreasOfNeedEntity = emptyList<AreaOfNeedEntity>()
 
```

Comment:
I think this kind of logic should be represented elsewhere, not in the service itself.

Reply:
any clues?

Reply:
maybe an extension function `merge(goal: Goal)` for the `GoalEntity`?

Reply:
so copy/paste this method into one in GoalEntity and call that from here?

Reply:
Could also be in a new file. 

e.g. `fun GoalEntity.merge(goal: Goal): GoalEntity {(code here but using 'this')}`

The `relatedAreasOfNeed` code could also be moved in there.

Reply:
Done in https://github.com/ministryofjustice/hmpps-sentence-plan/pull/84/commits/6a99769943e58504b9545fc1d880a4e2f44d6568